src/pool.rs

```diff
@@ -317,31 +330,16 @@ impl ConnectionPool {
                 let server = &*proxy;
                 let server_info = server.server_info();
 
-                if !server_infos.is_empty() {
-                    // Compare against the last server checked.
-                    if server_info != server_infos[server_infos.len() - 1] {
```

Comment:
I debated keeping this code in. It could be helpful to see in the logs. I opted to "do less" for this cut but it's easy to add back, so let me know!

Reply:
I haven't seen any uses for that piece of code yet. We can put it back someday if needed.